call-home.sh

```diff
@@ -0,0 +1,286 @@
+PERCONA_SEND_TIMEOUT=${PERCONA_SEND_TIMEOUT:-10}
+PERCONA_TELEMETRY_URL="${PERCONA_TELEMETRY_URL:-"https://127.0.0.1.nip.io/v1/telemetry/GenericReport"}"
+PERCONA_PRODUCT_FAMILY=${PERCONA_PRODUCT_FAMILY}
+PERCONA_PRODUCT_VERSION=${PERCONA_PRODUCT_VERSION}
+PERCONA_OPERATING_SYSTEM=${PERCONA_OPERATING_SYSTEM}
+PERCONA_DEPLOYMENT_METHOD=${PERCONA_DEPLOYMENT_METHOD}
+PERCONA_INSTANCE_ID=${PERCONA_INSTANCE_ID}
+
+json_message=""
+declare -A json_message_map
+declare -A metrics_map
+
+## USAGE
+usage()
+{
+    errorCode=${1:-0}
+
+    cat << EOF
+
+usage: $0 OPTIONS
+
+Collects telemetry information and sends it to a Percona server. The data collection
+may be disabled via setting an environment variable [PERCONA_TELEMETRY_DISABLE].
+
+The script validates only if mandatory parameters are provided.
+It does not validate the validity and itegritiy of provided parameters.
+
+The data will be sent to $PERCONA_TELEMETRY_URL in a JSON format.
+
+OPTIONS can be:
+
+  -h  Show this message
+
+  -f  [PERCONA_PRODUCT_FAMILY]          Product family identifier.                                  [REQUIRED]
+  -v  [PERCONA_PRODUCT_VERSION]         Product version.                                            [REQUIRED]
+  -s  [PERCONA_OPERATING_SYSTEM]        Operating system identifier.                                [REQUIRED]
+  -d  [PERCONA_DEPLOYMENT_METHOD]       Deployment method.                                          [REQUIRED]
+  -i  [PERCONA_INSTANCE_ID]             Instance id                                                 [Default: autogenerated]
+  -j  [PERCONA_INSTANCE_ID_FILE_PATH]   Path of the file where to store the unique ID of this node. [Default: $PERCONA_INSTANCE_ID_FILE_PATH]
+  -u  [PERCONA_TELEMETRY_URL]           Percona Telemetry Service endpoint                          [Default: $PERCONA_TELEMETRY_URL]
+  -t  [PERCONA_SEND_TIMEOUT]            Default timeout for the curl command.                       [Default: $PERCONA_SEND_TIMEOUT]
+
+For example,
+on a CentOS7, you may run the script as:
+
+./$0 -f "PRODUCT_FAMILY_MYSQL" -v "8.0.33" -s "\$(cat /etc/redhat-release)" -i "13f5fc62-35b4-4716-b3e6-96c761fc204d" -j /tmp/percona.telemetry -u "https://TO/DO/PROVIDE/ENDPOINT -t 1
+
+on Ubuntu, you may run the script as:
+./$0 -f "PRODUCT_FAMILY_MYSQL" -v "8.0.33" -s "\$(cat /etc/issue)" -i "13f5fc62-35b4-4716-b3e6-96c761fc204d" -j /tmp/percona.telemetry -u "https://TO/DO/PROVIDE/ENDPOINT -t 1
+
+EOF
+
+    if [[ $errorCode -ne 0 ]];
+    then
+        exit_script $errorCode
+    fi
+}
+
+# Perform any required cleanup and exit with the given error/success code
+exit_script()
+{
+    # Exit with a given return code or 0 if none are provided.
+    exit ${1:-0}
+}
+
+# Vaildate arguments to ensure that mandatory ones have been provided
+validate_args()
+{
+    local USAGE_TEXT="See usage for details."
+
+    if [[ -z "$PERCONA_PRODUCT_FAMILY" ]];
+    then
+        printf "PERCONA_PRODUCT_FAMILY not provided. %s\n" "$USAGE_TEXT" >&2
```

Comment:
```suggestion
        printf "PERCONA_PRODUCT_FAMILY is not provided. %s\n" "$USAGE_TEXT" >&2
```

and for the rest as well

Reply:
done